api/tests/test_lock_manager.py

```diff
@@ -0,0 +1,99 @@
+import asyncio
+import logging
+import os
+import sys
+import unittest
+from dataclasses import dataclass
+from typing import Optional
+from unittest import IsolatedAsyncioTestCase
+
+from core import logging
+from core.exceptions import NotFoundException
+from core.store.database import Database
+from core.util.value_holder import RequestIdHolder
+
+sys.path.insert(0, os.path.dirname(os.path.dirname(__file__)))
+from notd.lock_manager import LockManager
+from notd.lock_manager import LockTimeoutException
+from notd.store.retriever import Retriever
+from notd.store.saver import Saver
+
+openseaApiKey = os.environ['OPENSEA_API_KEY']
+
+async def test_function(lockManager: LockManager, index: int):
+    async with lockManager.with_lock(name='testing', timeoutSeconds=3, expirySeconds=1):
+        print(f'{index} got lock')
+        for i in range(20):
+            await asyncio.sleep(0.1)
+
+async def test_acquire_function(lockManager: LockManager, index: int, timeoutSeconds: Optional[int] = 6, expirySeconds: Optional[int] = 1 ):
+    print(f'{index} trying to acquire lock')
+    await lockManager.acquire_lock(name='testing', timeoutSeconds=timeoutSeconds, expirySeconds=expirySeconds)
+    for i in range(20):
+            await asyncio.sleep(0.1)
+
+class KibaAsyncTestCase(IsolatedAsyncioTestCase):
+
+    def __init__(self, methodName: str = 'runTest') -> None:
+        super().__init__(methodName=methodName)
+
+
+class LockManagerTestCase(KibaAsyncTestCase):
+
+    async def asyncSetUp(self) -> None:
+        requestIdHolder = RequestIdHolder()
+        name = os.environ.get('NAME', 'notd-api')
+        version = os.environ.get('VERSION', 'local')
+        environment = os.environ.get('ENV', 'dev')
+        isRunningDebugMode = environment == 'dev'
+
+        if isRunningDebugMode:
+            logging.init_basic_logging()
+        else:
+            logging.init_json_logging(name=name, version=version, environment=environment, requestIdHolder=requestIdHolder)
+
+        databaseConnectionString = Database.create_psql_connection_string(username=os.environ["DB_USERNAME"], password=os.environ["DB_PASSWORD"], host=os.environ["DB_HOST"], port=os.environ["DB_PORT"], name=os.environ["DB_NAME"])
+        self.database = Database(connectionString=databaseConnectionString)
+        self.saver = Saver(database=self.database)
+        self.retriever = Retriever(database=self.database)
+        self.lockManager = LockManager(retriever=self.retriever, saver=self.saver)
+        await self.database.connect()
+
+    async def asyncTearDown(self) -> None:
+        await self.database.disconnect()
+        await super().asyncTearDown()
+
+class TestLockExpiry(LockManagerTestCase):
+
+    # Test for Expiry
+    async def test_expiry(self):
+        await asyncio.gather(*[test_acquire_function(lockManager=self.lockManager, index=i, timeoutSeconds=5, expirySeconds=1) for i in range(1)])
```

Comment:
the range is useless, the gather is usless and even the function is useless

Reply:
in a test case you ALWAYS ALWAYS ALWAYS have the function being tested being used in the test

Reply:
questions: where is the function you are testing, and what are you actually testing